opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -1007,29 +1007,67 @@ namespace Opm
         }
 
         template<class Comm>
-        void communicateGroupRates(const Comm& comm) {
-            // sum over all nodes
-            for (auto& x : injection_group_rein_rates) {
-                comm.sum(x.second.data(), x.second.size());
-            }
+        void communicateGroupRates(const Comm& comm)
+        {
+            // Doing this separately since it is a single double, not a vector.
```

Comment:
why not just stick them at the front of the vector? does not matter if it's a single number or multiple, it's the act of communication that bites, not the summation.

Reply:
I can do that, just thought this was easier to understand.

Reply:
i think optimization should have higher priority than readability in this case.

Reply:
I'll change. I must say I am quite satisfied with the readability here as well though, combining lambdas is quite powerful.